src/v2/components/UI/Meter/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import styled from 'styled-components'
+import { themeGet } from 'styled-system'
+
+import Box from 'v2/components/UI/Box'
+
+const Container = styled(Box).attrs(({ p, borderColor, borderRadius }) => ({
+  border: '1px solid',
+  borderColor: borderColor,
+  borderRadius: borderRadius,
+  p: p,
+}))`
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  overflow: hidden;
+`
+
+const MeterProgress = styled(Box).attrs(({ borderRadius }) => ({
+  borderRadius: borderRadius,
+}))`
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  width: ${props => `${(props.amount / props.limit) * 100.0}%`};
+  ${props => `
+    background: linear-gradient(to right, ${themeGet(
+      `colors.${props.startColor}`
+    )(props)}, ${themeGet(`colors.${props.endColor}`)(props)});
+  `}
+`
+
+interface MeterProps {
+  startColor: string
+  endColor: string
+  amount: number
+  limit: number
+  borderRadius?: string
+  bg?: string
+  borderColor?: string
+  p?: number
+  mb?: number
```

Comment:
All common styled-system props can be imported via `@types/styled-system`: 

`import { SpaceProps, etc } from 'styled-system'`

```js
interface MeterProps extends SpaceProps {
  ...
}
```

Reply:
This maps 1:1 with available types: https://github.com/styled-system/styled-system/blob/master/docs/table.md

Reply:
++ thanks @damassi